lib/Parser/RegexParser.cpp

```diff
@@ -2188,9 +2199,23 @@ namespace UnifiedRegex
                 pendingCodePoint = NextChar();
             }
 
-            if (codePointToSet != INVALID_CODEPOINT)
+            if (codePointToSet != INVALID_CODEPOINT || prevprevWasACharSetAndPartOfRange)
             {
-                if (pendingRangeStart != INVALID_CODEPOINT)
+                if (unicodeFlagPresent && prevprevWasACharSetAndPartOfRange)
+                {
+                    Fail(JSERR_UnicodeRegExpRangeContainsCharClass);
```

Comment:
I believe this is redundant to the logic above. We would have already thrown SyntaxError before prevprevWasACharSetAndPartOfRange could be set to true.